### ISSUE TITLE
AttributeError in Chapter 3

### ISSUE THREAD

**Issue by afettouhi** [elegant-scipy/elegant-scipy#346]
```python
in_degrees = list(wormbrain.in_degree().values())
in_deg_distrib = np.bincount(in_degrees)
avg_in_degree = np.mean(in_degrees)
cumfreq = np.cumsum(in_deg_distrib) / np.sum(in_deg_distrib)
survival = 1 - cumfreq
```
this above code snippet from chapter 3 fails to run with up to date versions of numpy etc. I get a
`AttributeError: 'InDegreeView' object has no attribute 'values'`.

**Comment by stefanv** [comment]
Hi André, I cannot seem to replicate this issue.  Can you please report your versions of NumPy, SciPy, NetworkX, and Python?

**Comment by afettouhi** [comment]
```
numpy                     1.16.4           py37h7e9f1db_0  
numpy-base                1.16.4           py37hde5b4d6_0
networkx                  2.3                        py_0
python                    3.7.3                h0371630_0
scipy                     1.3.0            py37h7c811a0_0
```
using miniconda3 on Arch Linux.

**Comment by afettouhi** [comment]
Figured it out. I am using the code directly from the book and I can see from the notebooks you provide that `wormbrain.in_degree()` now is provided as a `dict` e.g. `dict(wormbrain.in_degree())`. After changing that it works as expected. Many thanks! Closing this.

**Comment by jni** [comment]
Ah, yes, apologies, this is because NetworkX released version 2.0 after we released the book. We corrected it online but book prints take years to be fixed! Thanks for reporting and feel free to ask more questions! =)

**Comment by afettouhi** [comment]
Nothing to apologize for. I had figured out that there was a change with the new NetworkX 2.0 package after sifting through parts of its documentation but I was doing it wrong. I kept changing the list function to dict instead of wrapping wormbrain.in_degree() in the dict function.
